libs/server-sdk/include/launchdarkly/server_side/bindings/c/all_flags_state/all_flags_state.h

```diff
@@ -0,0 +1,118 @@
+/** @file */
+// NOLINTBEGIN modernize-use-using
+
+#pragma once
+
+#include <launchdarkly/bindings/c/export.h>
+#include <launchdarkly/bindings/c/status.h>
+#include <launchdarkly/bindings/c/value.h>
+
+#ifdef __cplusplus
+extern "C" {  // only need to export C interface if
+// used by C++ source code
+#endif
+
+typedef struct _LDAllFlagsState* LDAllFlagsState;
+
+/**
+ * Frees an AllFlagsState.
+ * @param state The AllFlagState to free.
+ */
+LD_EXPORT(void) LDAllFlagsState_Free(LDAllFlagsState state);
+
+/**
+ * True if the LDAllFlagsState is valid.  False if there was
+ * an error, such as the data store being unavailable. When false, the other
+ * accessors will not return data.
+ *
+ * An invalid LDAllFlagsState can still be serialized successfully to a JSON
+ * string.
+ *
+ * @param state The LDAllFlagState to check for validity. Must not be NULL.
+ * @return True if the state is valid, false otherwise.
+ */
+LD_EXPORT(bool) LDAllFlagsState_Valid(LDAllFlagsState state);
+
+/**
+ * Serializes the LDAllFlagsState to a JSON string.
+ *
+ * This JSON is suitable for bootstrapping a client-side SDK.
+ *
+ * @param state The LDAllFlagState to serialize. Must not be NULL.
+ * @return A JSON string representing the LDAllFlagsState. The caller must free
+ * the string using LDMemory_FreeString.
+ */
+LD_EXPORT(char*)
+LDAllFlagsState_SerializeJSON(LDAllFlagsState state);
+
+/**
```

Comment:
This whole comment points to the danger of this kind of API. 

Theoretically we could create a new typedef `LDValueReference` for these situations. And all the bindings. But it wouldn't have a binding for `_Free`. That'd prevent at least one error. 

Another thing we could do is name this `_ValueReference` or something, rather than `_Value`.